dotnet/src/Connectors/Connectors.Amazon/Bedrock/Core/Models/AI21Labs/AI21JambaIOService.cs

```diff
@@ -0,0 +1,195 @@
+
+/// <summary>
+/// Input-output service for AI21 Labs Jamba model.
+/// </summary>
+internal sealed class AI21JambaIOService : IBedrockModelIOService
+{
+    /// <summary>
+    /// Builds InvokeModel request Body parameter with structure as required by AI21 Labs Jamba model.
+    /// </summary>
+    /// <param name="modelId">The model ID to be used as a request parameter.</param>
+    /// <param name="prompt">The input prompt for text generation.</param>
+    /// <param name="executionSettings">Optional prompt execution settings.</param>
+    /// <returns></returns>
+    object IBedrockModelIOService.GetInvokeModelRequestBody(string modelId, string prompt, PromptExecutionSettings? executionSettings)
+    {
+        var exec = AmazonJambaExecutionSettings.FromExecutionSettings(executionSettings);
+        List<AI21JambaRequest.AI21TextGenerationRequest.Msg> messages = new()
+        {
+            new AI21JambaRequest.AI21TextGenerationRequest.Msg()
+            {
+                Role = "user",
+                Content = prompt
+            }
+        };
+
+        // Get the prompt execution settings from ExtensionData dictionary of PromptExecutionSettings or AmazonJambaTextExecutionSettings specific parameters.
+        var requestBody = new AI21JambaRequest.AI21TextGenerationRequest()
+        {
+            Messages = messages,
+            Temperature = BedrockModelUtilities.GetExtensionDataValue<float?>(exec.ExtensionData, "temperature") ?? exec.Temperature,
+            TopP = BedrockModelUtilities.GetExtensionDataValue<float?>(exec.ExtensionData, "top_p") ?? exec.TopP,
+            MaxTokens = BedrockModelUtilities.GetExtensionDataValue<int?>(exec.ExtensionData, "max_tokens") ?? exec.MaxTokens,
+            Stop = BedrockModelUtilities.GetExtensionDataValue<IList<string>?>(exec.ExtensionData, "stop") ?? exec.Stop,
+            NumberOfResponses = BedrockModelUtilities.GetExtensionDataValue<int?>(exec.ExtensionData, "n") ?? exec.NumberOfResponses,
+            FrequencyPenalty = BedrockModelUtilities.GetExtensionDataValue<double?>(exec.ExtensionData, "frequency_penalty") ?? exec.FrequencyPenalty,
+            PresencePenalty = BedrockModelUtilities.GetExtensionDataValue<double?>(exec.ExtensionData, "presence_penalty") ?? exec.PresencePenalty
+        };
+
+        return requestBody;
+    }
+
+    /// <summary>
+    /// Extracts the test contents from the InvokeModelResponse as returned by the Bedrock API.
+    /// </summary>
+    /// <param name="response">The InvokeModelResponse object provided by the Bedrock InvokeModelAsync output.</param>
+    /// <returns></returns>
+    IReadOnlyList<TextContent> IBedrockModelIOService.GetInvokeResponseBody(InvokeModelResponse response)
+    {
+        using var memoryStream = new MemoryStream();
+        response.Body.CopyToAsync(memoryStream).ConfigureAwait(false).GetAwaiter().GetResult();
+        memoryStream.Position = 0;
+        using var reader = new StreamReader(memoryStream);
```

Comment:
Similar to the other comment, avoid the allocation of an extre `MemoryStream`, apply the suggestion to all `IOServices`.